### ISSUE TITLE
Nested dynamic blocks flagged as errors

### ISSUE THREAD

**Issue by ephraim-gr** [hashicorp/vscode-terraform#1877]
### Extension Version

v2.33.0

### VS Code Version

Version: 1.95.1
Commit: 65edc4939843c90c34d61f4ce11704f09d3e5cb6
Date: 2024-10-31T05:14:54.222Z
Electron: 32.2.1
ElectronBuildId: 10427718
Chromium: 128.0.6613.186
Node.js: 20.18.0
V8: 12.8.374.38-electron.0
OS: Darwin arm64 23.6.0

### Operating System

macOS Sonoma 14.7.1 (23H222)

### Terraform Version

Terraform v1.9.8 on darwin_arm64

### Steps to Reproduce

Basically the same as https://github.com/hashicorp/vscode-terraform/issues/1594.
I specifically am trying to do this and get the same error `Unexpected block: Blocks of type "dynamic" are not expected here`:
```
dynamic "global_secondary_index" {
    for_each = var.global_secondary_indexes
    content {
      name               = global_secondary_index.key
      hash_key           = global_secondary_index.value.hash_key
      range_key          = global_secondary_index.value.range_key
      projection_type    = global_secondary_index.value.projection_type != null ? global_secondary_index.value.projection_type : global_secondary_index.value.non_key_attributes != null ? "INCLUDE" : "ALL"
      non_key_attributes = global_secondary_index.value.non_key_attributes
      read_capacity      = var.billing.mode == "PROVISIONED" ? global_secondary_index.value.read_capacity : null
      write_capacity     = var.billing.mode == "PROVISIONED" ? global_secondary_index.value.write_capacity : null
      dynamic "on_demand_throughput" {
        for_each = var.billing.mode == "PAY_PER_REQUEST" ? [var.billing] : []
        content {
          max_read_request_units  = on_demand_throughput.value.max_read_request_units
          max_write_request_units = on_demand_throughput.value.max_write_request_units
        }
      }
    }
  }
  ```

### Expected Behavior

Shouldn't give me this error as it was fixed in version [2.29.0](https://github.com/hashicorp/vscode-terraform/releases/tag/v2.29.0) as was stated in this [comment](https://github.com/hashicorp/vscode-terraform/issues/1594#issuecomment-1802258565)

### Actual Behavior

Still get the error

### Help Wanted

- [ ] I'm interested in contributing a fix myself

### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by dbanck** [comment]
Hi @ephraim-gr,

Thanks for the report! To reproduce the issue, I need more information about your configuration. Which data/resource/provider has a `global_secondary_index`?

**Comment by ephraim-gr** [comment]
Hi @dbanck, thanks for the response.
I'm using [aws dynamodb table resource](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/dynamodb_table#global_secondary_index)

**Comment by dbanck** [comment]
Thanks! I was able to reproduce the error you're seeing.

`on_demand_throughput` was added to the AWS provider in 5.72.0. The latest version of the language server shipped with version 5.71.0, so it doesn't know about the new block yet.

We bundle the latest provider schema available at the time of the language server and extension release. If new attributes, blocks or resources are introduced since then, we can't automatically reflect them (yet). The next language server release will bundle the updated schema and fix the problem. We apologize for the inconvenience.

In the meantime, when you run `terraform init`, we'll use the appropriate provider schema from disk.

In case you want to learn more about how we handle provider schemes, we have an explainer here: https://github.com/hashicorp/terraform-ls/blob/main/docs/schema.md

**Comment by ephraim-gr** [comment]
Got it, thank you very much!

**Comment by dbanck** [comment]
@ephraim-gr  We have released a new version [2.34.0](https://github.com/hashicorp/vscode-terraform/releases/tag/v2.34.0). This updates the schema to the latest version and should fix the validation error. The update should show up automatically in VS Code.

--- 

If you experience any other validation related bug, please let us know via a new issue.
